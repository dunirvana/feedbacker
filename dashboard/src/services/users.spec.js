import mockAxios from 'axios'
import UsersService from './users'

jest.mock('axios')

describe('UsersService', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })
 
  it('should return data from user', async () => {
    const user = {
        id: '99998888',
        name: 'Teste',
        email: 'teste@email.com',
        apiKey: '1234.5678.9012',
        createdAt: '1599264000000'
      }

    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await UsersService(mockAxios).getMe()
    expect(response.data).toHaveProperty('id')
    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('email')
    expect(response.data).toHaveProperty('apiKey')
    expect(response.data).toHaveProperty('createdAt')
    expect(response).toMatchSnapshot()
  })  

  it('should generate api key', async () => {
    const api = {
        apiKey: '9999.7777.9012'
      }

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: api })
    })

    const response = await UsersService(mockAxios).generateApiKey()
    expect(response.data).toHaveProperty('apiKey')
    expect(response).toMatchSnapshot()
  })  
})

