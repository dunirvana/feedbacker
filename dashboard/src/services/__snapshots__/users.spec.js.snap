// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersService should generate api key 1`] = `
Object {
  "data": Object {
    "apiKey": "9999.7777.9012",
  },
}
`;

exports[`UsersService should return data from user 1`] = `
Object {
  "data": Object {
    "apiKey": "1234.5678.9012",
    "createdAt": "1599264000000",
    "email": "teste@email.com",
    "id": "99998888",
    "name": "Teste",
  },
}
`;
